x-pack/plugins/security_solution/server/endpoint/services/actions.ts

```diff
@@ -239,7 +240,8 @@ export const getPendingActionCounts = async (
     });
 
     const pendingActions: EndpointAction[] = recentActions.filter((action) => {
-      return ackResponseActionIdList.includes(action.action_id) // if has ack
+      return isPendingActionResponsesWithAckEnabled &&
+        ackResponseActionIdList.includes(action.action_id) // if has ack
```

Comment:
so when this flag is _not_ enabled.. it goes the current/old action route.

What is the impact if the Endpoint binary is still using the "new" pattern? I guess it still sends the same sort of fleet response, and we just call that done as we somewhat already do. Are there any concerns with this new combination? (i.e. Endpoints using the new-action pattern, but kibana treating it as the old-action-pattern)

Reply:
@pzl So, when the flag is set to `false` or when we use the old logic for pending actions, there could be that the endpoint is using the new action index for isolation requests, and if writing to it fails for some reason, it does create a response in the new response index. These (action, response) would just show up in the activity log as a failed action/response. I don't think this is particularly concerning even though the pending action is using the old logic. Not sure if this is what you had in mind? 🤔

Reply:
@pzl so I tested this manually and I found that it doesn't work as it should. The agent still waits for the new response index doc to update statuses. I'm going to fix this so that we don't show any pending status for 7.16 endpoints when this flag is off.

Reply:
@ashokaditya - this makes sense to me.  I think the feature flag for pending actions should turn it entirely OFF regardless of old logic vs new or Endpoint version.  It would be confusing for users to see logic for some, but not others.

Reply:
done acab278e2e6cc8b502619c7a9af6a8c4a46d5c89